examples/dev/triangle_edge.py

```diff
@@ -0,0 +1,259 @@
+import typing
+
+import numba
+import numpy as np
+
+import napari
+from napari.layers import Points, Shapes, Vectors
+from napari.layers.shapes._shapes_utils import generate_2D_edge_meshes
+
+
+def generate_regular_polygon(n, radius=1):
+    """
+    Generates the vertices of a regular n-sided polygon centered at the origin.
+
+    Parameters:
+    n (int): The number of sides (vertices).
+    radius (float): The radius of the circumscribed circle. Default is 1.
+
+    Returns:
+    np.ndarray: An array of shape (n, 2) containing the vertices coordinates.
+    """
+    angles = np.linspace(0, 2 * np.pi, n, endpoint=False)
+    return np.column_stack((radius * np.cos(angles), radius * np.sin(angles)))
+
+
+# @numba.njit
+def generate_order_vectors(path_, closed) -> np.ndarray:
+    """Generate the vectors tangent to the path."""
+    n = path_[1:] - path_[:-1]
+    norm = np.linalg.norm(n, axis=1, keepdims=True)
+    normals = n / norm
+    vec = np.empty((path_.shape[0], 2, 2))
+    vec[:, 0] = path_
+    vec[:-1, 1] = normals
+    if closed:
+        vec[-1, 1] = (path_[0] - path_[-1]) / np.linalg.norm(path_[-1] - path_[0])
+    else:
+        vec[-1, 1] = -vec[-2, 1]
+    return vec
+
+# @numba.njit
+def generate_miter_helper_vectors(normal_vector: np.ndarray) -> np.ndarray:
+    """Generate the miter helper vectors.
+
+    for each point on the path, the miter helper vectors are pairs of vectors
+    First vector is the normal vector scaled by 1/2
+    Second vector is the normal vector of previous point scaled by -1/2
+    """
+    vec2 = normal_vector.copy()
+    vec2[:, 1] *= 0.5
+    vec3 = vec2.copy()
+    vec3[:, 1] *= -1
+    vec3[:, 1] = np.roll(vec3[:, 1], 1, axis=0)
+    vec3[:, 0] += vec2[:, 1]
+    return np.concatenate([vec2, vec3], axis=0)
+
+@numba.njit
+def generate_orthogonal_vectors(normal_vector: np.ndarray) -> np.ndarray:
+    """Generate the orthogonal vectors to the normal vectors.
+
+    The orthogonal vector starts at the middle of the normal vector and is orthogonal to it.
+    It has length equal to half of the normal vector.
+    """
+    vec2 = normal_vector.copy()
+    vec2[:, 1] *= 0.5
+    vec5 = vec2.copy()
+    vec5[:, 0] += vec2[:, 1]
+    vec5[:, 1, 0] = -vec2[:, 1, 1]
+    vec5[:, 1, 1] = vec2[:, 1, 0]
+    return vec5
```

Comment:
Since this is an example, perhaps:
```suggestion
    vec1 = normal_vector.copy()
    vec1[:, 1] *= 0.5
    orthogonal_vector = vec1.copy()
    
    orthogonal_vector[:, 0] += vec1[:, 1]
    orthogonal_vector[:, 1, 0] = -vec1[:, 1, 1]
    orthogonal_vector[:, 1, 1] = vec1[:, 1, 0]
    return orthogonal_vector
```